cgi-bin/Apache/LiveJournal.pm

```diff
@@ -499,7 +500,7 @@ sub trans
         # -- uppercase usernames
         # -- users with hyphens/underscores, except users from external domains (see table 'domains')
         if ( $orig_user ne lc($orig_user) ||
-            $orig_user =~ /[_-]/ && $u && $u->journal_base !~ m!^http://$host!i && $opts->{'vhost'} !~ /^other:/) {
+            $orig_user =~ /[_-]/ && $u && $u->journal_base !~ m!^https?://$host!i && $opts->{'vhost'} !~ /^other:/) {
```

Comment:
Hmm, I was concerned about making sure we allow both http and https, but $u->journal_base is already going to be either http or https based on the page it's in so that was actually an invalid concern.